src/dft/backends/mklcpu/commit.cpp

```diff
@@ -25,25 +25,107 @@
 
 #include "oneapi/mkl/types.hpp"
 #include "oneapi/mkl/dft/types.hpp"
+#include "oneapi/mkl/dft/descriptor.hpp"
 
 #include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
 
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "mkl_service.h"
+#include "mkl_dfti.h"
+
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklcpu {
 
-void commit_f(descriptor<precision::SINGLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+template <precision prec, domain dom>
+class commit_derived_impl : public detail::commit_impl {
+public:
+    commit_derived_impl(sycl::queue queue, dft_values config_values)
+            : detail::commit_impl(queue),
+              status(-1) {
+        if (config_values.rank == 1) {
+            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom),
+                                          config_values.rank, config_values.dimensions[0]);
+        }
+        else {
+            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom),
+                                          config_values.rank, config_values.dimensions.data());
+        }
+        if (status != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft", "commit", "DftiCreateDescriptor failed");
+        }
+
+        set_value(handle, config_values);
+
+        status = DftiCommitDescriptor(handle);
+        if (status != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft", "commit", "DftiCommitDescriptor failed");
+        }
+    }
+
+    commit_derived_impl(const commit_derived_impl* other) : detail::commit_impl(*other) {}
```

Comment:
I think we still need to be careful here with double-freeing a handle.

Reply:
lets get rid of this copy constructor unless a explicit need arises